subjects/java/checkpoints/circular-linkedlist/README.md

```diff
@@ -0,0 +1,118 @@
+## Circular Linked List
+
+### Instructions
+
+Create a circular single linked list data structure that implements the following methods:
+
+- `at(int index)`: to access an element by its index.
+- `add(int value)`: to add an element at the end of the list.
+- `remove(int index)`: to remove an element by its index.
+- `size()`: to return the size of the list.
+
+Define these methods in an interface called `LinkedList`, and implement this interface in a class that you will create. Additionally, add a private method `next(Node node)` in your class that will be used in the other methods. This method should print the message "Go to next node" each time it is called.
```

Comment:
As for the `SingleLinkedList` - I find this a bit confusing. What I understood looking at the whole subject is the following:

```
Given the provided interfaced `LinkedList`, implement the class `CircularLinkedList` with the elements defined below. Additionally ...
```

Reply:
i will rephrase it.